src/push_service/cdn.rs

```diff
@@ -90,26 +116,187 @@ impl PushService {
         .map_err(Into::into)
     }
 
-    /// Upload attachment to CDN
-    ///
-    /// Returns attachment ID and the attachment digest
-    pub async fn upload_attachment(
+    #[tracing::instrument(skip(self), level=tracing::Level::TRACE)]
+    pub(crate) async fn get_attachment_resumable_upload_url(
         &mut self,
-        attrs: AttachmentV2UploadAttributes,
-        mut reader: impl Read + Send,
-    ) -> Result<(u64, Vec<u8>), ServiceError> {
-        let attachment_id = attrs.attachment_id;
-        let mut digester =
-            crate::digeststream::DigestingReader::new(&mut reader);
+        attachment_upload_form: &AttachmentUploadForm,
+    ) -> Result<Url, ServiceError> {
```

Comment:
The attachment_upload_form will be quite noisy in tracing; should we have some Display impl like I'm doing in #335 ?

Reply:
I think it's ok, it's not like we upload attachments like crazy anyways.

Reply:
Maybe you don't! ;-)